docs/development/architecture.md

```diff
@@ -0,0 +1,180 @@
+
+A few key dependencies:
+- **Golang**: Written using the [golang]() language. 
+- **Gorilla**: is the web framework we use
+- **Kubernetes and Helm**: Uses Kubernetes and Helm golang packages.
+
+Packages are stored in the [backend/pkg](https://github.com/headlamp-k8s/headlamp/blob/main/backend/pkg/) folder. The packages available are cache, config, helm, kubeconfig, logger, plugins, portforward, and utils. Future code should be organized in packages. There is still some code inside cmd/ folder that probably should be migrated to separate packages.
+
+For each module we have a test file. For example config.go comes with config_test.go.
+
+### App
+
+- The [app/](https://github.com/headlamp-k8s/headlamp/tree/main/app) folder contains everything to do with the app.
+- Entry point is [app/electron/main.ts](https://github.com/headlamp-k8s/headlamp/blob/main/app/electron/main.ts)
+
+The app depends on the frontend/ and the backend/. 
+
+A few key dependencies:
+**TypeScript**: Written in TypeScript.
+**Electron**: Used to create a native desktop application for Headlamp.
+**Yargs**: Used for command line argument processing.
+
+### Plugins
+
+- The [plugins/headlamp-plugin](https://github.com/headlamp-k8s/headlamp/tree/main/plugins/headlamp-plugin) folder contains the headlamp-plugin tool which provides plugin build and management tools.
+- [plugins/examples](https://github.com/headlamp-k8s/headlamp/tree/main/plugins/examples) folder contains some example plugins.
+- [frontend/src/plugin](https://github.com/headlamp-k8s/headlamp/tree/main/frontend/src/plugin) folder contains frontend code related to plugins.
+- [backend/pkg/plugins/](https://github.com/headlamp-k8s/headlamp/tree/main/backend/pkg/plugins/) folder contains backend code related to plugins.
+- [headlamp-k8s/plugins repo](https://github.com/headlamp-k8s/plugins) contains plugins maintained by the Headlamp team. Including plugins for projects like Flux, Backstage and Inspektor Gadget.
+- See the [Plugin Development Documentation](https://headlamp.dev/docs/latest/development/plugins/) for more information about developing plugins.
+
+A few key dependencies:
+**TypeScript**: Written in TypeScript.
+
+## Design Decisions
+
+- **Go for Backend**: Selected because Kubernetes APIs and ecosystem APIs use golang. Also for its performance and concurrency capabilities. 
+- **React for Frontend**: Chosen for its component-based architecture and strong community support.
+- **Extensibility**: Designed with a plugin system to allow easy customization and extension.
+- **Multi-Cluster Support**: Designed to manage multiple Kubernetes clusters from a single interface.
+- **Realtime**: Designed to react in real time to changes in clusters.
+
+We ask that people write issues proposing large changes. See more in the [contribution guide](https://headlamp.dev/docs/latest/contributing).
```

Comment:
This seems out of context in this subsection or even section. Belongs to contributing.